src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -45,10 +46,37 @@ public RedisCache(IOptions<RedisCacheOptions> optionsAccessor)
                 throw new ArgumentNullException(nameof(optionsAccessor));
             }
 
-            _options = optionsAccessor.Value;
+            RedisCacheOptions options = optionsAccessor.Value;
+ 
+             // This allows partitioning a single backend cache for use with multiple apps/services.
+            _instance = options.InstanceName ?? string.Empty;
 
-            // This allows partitioning a single backend cache for use with multiple apps/services.
-            _instance = _options.InstanceName ?? string.Empty;
+            if (options.ConnectionFactory != null)
+            {
+                _asyncConnectionFactory = options.ConnectionFactory;
+                _connectionFactory = () => _asyncConnectionFactory().Result;
```

Comment:
Is sync-over-async appropriate here? Is it possible this will cause problems?

Reply:
i don't see much problem here. this sync factory will be called by sync methods only. 
either we do this or we have a async-over-sync, which is less desirable IMO.

Reply:
> i don't see much problem here

See https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html, and especially the `Resources` section towards the bottom. It explains why blocking on an async operation leads to issues.

Previously, when calling the sync methods, it will call `ConnectionMultiplexer.Connect` (the sync method) which is the right thing to do. Now we only have a `Func<Task<IConnectionMultiplexer>> ConnectionFactory;` async method, which is not right to call from the synchronous path.

Reply:
How about adding a IConnectionMultiplexer field in RedisCacheOptions?